action.yml

```diff
@@ -65,8 +67,28 @@ runs:
       uses: actions/checkout@v4
       with:
         repository: ${{ inputs.pmd-ruleset-repo }}
-        ref: ${{ inputs.pmd-ruleset-ref }}
         path: pmd-ruleset
+        fetch-depth: 0
+
+    - name: Determine ruleset ref to checkout
+      id: find-ruleset-ref
+      run: |
+        set -x;
+        cd pmd-ruleset;
+        if [[ -z "${{ inputs.pmd-ruleset-ref-override }}" ]]; then
+          echo "No ruleset ref override provided, finding latest tag starting with PMD version: ${{ inputs.pmd-version }}";
+          ruleset_ref="$(git tag --list --sort=-taggerdate "${{ inputs.pmd-version }}*" | head -1)";
```

Comment:
maybe you can push an additional mutable tag at release time of the ruleset repo with just the pmd version - so you don't need any additional logic here

Reply:
The ruleset repo doesn't necessarily need updating for a new version of PMD (e.g. PMD 7.2.0 contained no new rules, so we would have only needed to update either `ya-pmd-scan` or the individual components that wanted to consume it).